### PR TITLE
Civ 5313 postcode validation

### PR DESCRIPTION
**Pull request by MMNycz**
### JIRA link

https://tools.hmcts.net/jira/browse/CIV-5313

### Change description

Update defendant postcode validation

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [X] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/main/app/forms/validation/validators/postCodeNotInScotlandOrNI.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { ValidationArguments, ValidatorConstraint, ValidatorConstraintInterface } from "@hmcts/class-validator"
+import { ValidationErrors } from "forms/models/address"
+
+@ValidatorConstraint({ name: 'postcodeNotInScotlandOrNI', async: false })
+export class PostcodeNotInScotlandOrNIValidator implements ValidatorConstraintInterface {
```
Comment (sabahirfan) [inline src/main/app/forms/validation/validators/postCodeNotInScotlandOrNI.ts:5]: Please add unit tests for this validator. Example here: https://github.com/hmcts/cmc-validators/blob/master/src/test/isValidPostcode.spec.ts

File: src/main/public/js/postcode-lookup.js [hunk 2]
```diff
@@ -18,9 +18,32 @@
...
+
+            if (isUKPostcode(enteredPostcode) && !isScotlandOrWalesPostcode(enteredPostcode)) {
+              lookupPostcode(enteredPostcode, postcodeLookupWidget);
+            } else {
+              displayPostcodeError(postcodeLookupWidget);
+            }
           })
 
+        function isUKPostcode(postcode) {
+          var ukPostcodePattern = /^[A-Z]{1,2}\d[A-Z\d]? \d[A-Z]{2}$/i;
```
Comment (sabahirfan) [inline src/main/public/js/postcode-lookup.js:31]: I think this is the regex for UK postcode validation: `/^(([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z]))))[0-9][A-Za-z]{2}))$/ `

Comment (MMNycz) [reply src/main/public/js/postcode-lookup.js:31]: I check that in documents you send to me - thank you for that

File: src/main/app/forms/validation/validators/postCodeNotInScotlandOrNI.ts (hunk 1 above)
Comment (MMNycz) [reply src/main/app/forms/validation/validators/postCodeNotInScotlandOrNI.ts:5]: added

**Review by sabahirfan — APPROVED** [review]

**Review by asthamalviya — APPROVED** [review]

**Pull request merged**